### PR TITLE
[WIP] Edit bar

### PR DESCRIPTION
**Pull request by lewiscollard**
So here is a thing, experimental and possibly incomplete, to add an 'edit this thing' link. If you are on a detail page (i.e. `object` is in the template context), it'll give you an edit link if you are logged in as an admin. Otherwise, it'll give a link to edit the current page.

Testing this on TTP to work out any issues (see 102 over there), let's see how this works out. :)

### CODE REVIEW

**Comment by danielsamuels** [comment]
I've got some thoughts on this, but I want to spend some time hacking around on it.

**Comment by danielsamuels** [comment]
I've been playing with this a little on the TTP project and I've come up with the following iteration:



I think I prefer the bar at the top, as well as the ability to add another item alongside the existing edit link.

What do you think?

**Comment by dan-gamble** [comment]
LGTM we might as well basically call ourself Django CMS now boys

**Review by danielsamuels — APPROVED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/templatetags/site.py [hunk 1]
```diff
@@ -196,7 +198,67 @@ def md(value, inline=True):
...
+
+@library.global_function
+@library.render_with('edit_bar.html')
+@jinja2.contextfunction
+def edit_bar(context):
+    context = dict(context)
+    request = context['request']
+
+    # Don't show to non-admins, and don't pretend that /search/ is editable.
+    if not request.user.is_staff or request.path == '/search/':
```
Comment (dan-gamble) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/templatetags/site.py:213]: Should this take in to account page edit permissions as well?

**Review by dan-gamble — COMMENTED** [review]
Has the edit bar had responsive done on it? Currently it looks like it doesn't even hide on mobile

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/templatetags/site.py (hunk 1 above)
Comment (danielsamuels) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/templatetags/site.py:213]: It does on line 243.

Comment (dan-gamble) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/templatetags/site.py:213]: Whoops, misread what this line does 👍

**Pull request merged**